cli/src/main.rs

```diff
@@ -144,28 +147,15 @@ async fn new_instance(
         vcpus,
     };
 
-    let mut disks: Vec<DiskRequest> =
-        Vec::with_capacity(crucible_target.len() / 3);
-
-    let mut disk_i = 0;
-    for target_group in crucible_target.chunks(3) {
-        assert_eq!(target_group.len(), 3);
-        disks.push(DiskRequest {
-            name: format!("d{}", disk_i),
-            address: target_group.to_vec(),
-            slot: Slot(disk_i),
-            read_only: false,
-            key: None,
-            gen: 0,
-        });
-        disk_i += 1;
+    for disk in &disks {
+        assert_eq!(disk.address.len(), 3);
```

Comment:
Should this perhaps be part of the `parse_crucible_disks`, and then the rest of the validation be left to the API we're calling into?

Reply:
Moved target validation server side in 00009e6